_drafts/fwup-signing.md

```diff
@@ -0,0 +1,639 @@
+---
+title: "Authenticate firmware builds with code signing"
+description: WIP
+author: francois
+---
+
+<!-- excerpt start -->
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+{% include toc.html %}
+
+## Firmware Signing Explained
+
+### What is code signing
+
+Code signing is a method of proving a file was created by a trusted source and
+has not been tampered with. This is achieved by creating a signature for the
+file: a token which can be verified but not forged.
+
+We all use signatures in our day to day life, by scribbling our name at the
+bottom of our credit card receipts, contracts, and bank checks. These signatures
+can be verified quite easily by comparing them to previous signatures, but are
+hard to forge because they rely on an individual's muscle memory.
+
+Code signing uses cryptographic algorithms to achieve a similar goal. The
+resulting signatures can only be generated by people who know a given password,
+also known as a secret key, but can be verified as genuine by anyone.
+
+Additionally, the signatures are tied to a given firmware binary so should the
+binary change by so much as 1 bit the signature would no longer be valid.
+
+### Why sign our firmware
+
+By implementing signature verification in our bootloader we can identify
+whether or not a given firmware update was provided by the manufacturer, or if
+it has been tampered with. The bootloader can then decide to either warn the
+user, void the device's warranty, or simply refuse to run the unauthenticated
+binary.
+
+With more and more devices connected to the internet, security is an
+increasingly hot topic in firmware development. A device which accepts firmware
+updates over the wireless or internet connectivity but does not verify it opens
+itself to compromise. By feeding it with a malicious firmware image, an attacker
+might:
+* Brick the device, or the whole fleet
+* Snoop on end users and compromise their privacy and security
+* Strategically malfunction at a critical time
+
+These are highly undesirable outcomes, which can be effected at scale due to the
+internet of things. In 2020, it is reckless to implement firmware update for our
+systems without some form of authentication.
+
+> **What signing is not**: code signing is an important component of firmware
```

Comment:
Maybe even move it to its own section rather than a big block comment.

Reply:
I don't think I have quite enough for a section